python/foglamp/services/core/api/update.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" FogLAMP package updater API support"""
+
+from aiohttp import web
+import uuid
+import asyncio
+import json
+
+# FogLAMP imports
+from foglamp.services.core import server
+from foglamp.common import logger
+
+_LOG_LEVEL_DEBUG = 10
+_logger = logger.setup(__name__, level=_LOG_LEVEL_DEBUG)
+
+__author__ = "Massimiliano Pinto"
+__copyright__ = "Copyright (c) 2018 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_help = """
+    -----------------------------------------
+    | PUT             | /foglamp/update     |
+    -----------------------------------------
+    | no paylod       |                     |
+    -----------------------------------------
+"""
+
+###
+# The update.py is part of the FogLAMP REST API:
+# when called via PUT /foglamp/update it will add the "update task"
+# to scheduler queue for execution.
+#
+# This only happens if the 'FogLAMPUpdater' is found
+# and the scheduled task is enabled.
+#
+
+_FOGLAMP_UPDATE_TASK = "FogLAMPUpdater"
+
+async def update_package(request):
+    """ Queues the execution of FogLAMP package update task
+
+    :Example: curl -X PUT http://localhost:8081/foglamp/update
```

Comment:
endpoint should be `http://localhost:8081/foglamp/snap/update` ?

Reply:
API specs is generic, so /foglamp/update


That will work with any kind of update mechanism.

The task scripts/task/update is copied by package manager in the right place and it contains package type specialisation with logging, names etc